### PR TITLE
Add `InkResponse`, `Material` and fix `Opacity`

### PR DESCRIPTION
**Pull request by peixinli**
Related issue: #141668

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/rfw/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 1.0.24
+* Adds `InkResponse` material widget.
+* Adds `Material` material widget.
+* Fixes the `Opacity` core widget.
```
Comment (Hixie) [inline packages/rfw/CHANGELOG.md:4]: we should document what was fixed

File: packages/rfw/lib/src/flutter/material_widgets.dart [hunk 2]
```diff
@@ -337,6 +340,42 @@ Map<String, LocalWidgetBuilder> get _materialWidgetsDefinitions => <String, Loca
...
+      hoverColor: ArgumentDecoders.color(source, ['hoverColor']),
+      highlightColor: ArgumentDecoders.color(source, ['highlightColor']),
+      splashColor: ArgumentDecoders.color(source, ['splashColor']),
+      enableFeedback: source.v<bool>(['enableFeedback']) ?? true,
+      excludeFromSemantics: source.v<bool>(['excludeFromSemantics']) ?? false,
+      canRequestFocus: source.v<bool>(['canRequestFocus']) ?? true,
+      onFocusChange: source.handler(['onFocusChange'], (VoidCallback trigger) => (bool focus) => trigger()),
+      autofocus: source.v<bool>(['autofocus']) ?? false,
+      hoverDuration:
+          ArgumentDecoders.duration(source, ['hoverDuration'], context),
```
Comment (Hixie) [inline packages/rfw/lib/src/flutter/material_widgets.dart:374]: weird wrapping

File: packages/rfw/lib/src/flutter/material_widgets.dart [hunk 3]
```diff
@@ -337,6 +340,42 @@ Map<String, LocalWidgetBuilder> get _materialWidgetsDefinitions => <String, Loca
...
+      onSecondaryTapUp: source.handler(['onSecondaryTapUp'], (VoidCallback trigger) => (TapUpDetails details) => trigger()),
+      onSecondaryTapDown: source.handler(['onSecondaryTapDown'], (VoidCallback trigger) => (TapDownDetails details) => trigger()),
+      onSecondaryTapCancel: source.voidHandler(['onSecondaryTapCancel']),
+      onHighlightChanged: source.handler(['onHighlightChanged'], (VoidCallback trigger) => (bool highlighted) => trigger()),
+      onHover: source.handler(['onHover'], (VoidCallback trigger) => (bool hovered) => trigger()),
+      containedInkWell: source.v<bool>(['containedInkWell']) ?? false,
+      highlightShape: ArgumentDecoders.enumValue<BoxShape>(BoxShape.values, source, ['highlightShape']) ?? BoxShape.circle,
+      radius: source.v<double>(['radius']),
+      borderRadius: ArgumentDecoders.borderRadius(source, ['borderRadius'])
+          ?.resolve(Directionality.of(context)),
```
Comment (Hixie) [inline packages/rfw/lib/src/flutter/material_widgets.dart:362]: weird wrapping

**Review by Hixie — APPROVED** [review]
LGTM modulo minor issues

File: packages/rfw/CHANGELOG.md (hunk 1 above)
Comment (peixinli) [reply packages/rfw/CHANGELOG.md]: Updated.

File: packages/rfw/lib/src/flutter/material_widgets.dart (hunk 2 above)
Comment (peixinli) [reply packages/rfw/lib/src/flutter/material_widgets.dart]: Fixed.

File: packages/rfw/lib/src/flutter/material_widgets.dart (hunk 3 above)
Comment (peixinli) [reply packages/rfw/lib/src/flutter/material_widgets.dart]: Fixed.

File: packages/rfw/lib/src/flutter/material_widgets.dart [hunk 4]
```diff
@@ -337,6 +340,42 @@ Map<String, LocalWidgetBuilder> get _materialWidgetsDefinitions => <String, Loca
...
+      highlightColor: ArgumentDecoders.color(source, ['highlightColor']),
+      splashColor: ArgumentDecoders.color(source, ['splashColor']),
+      enableFeedback: source.v<bool>(['enableFeedback']) ?? true,
+      excludeFromSemantics: source.v<bool>(['excludeFromSemantics']) ?? false,
+      canRequestFocus: source.v<bool>(['canRequestFocus']) ?? true,
+      onFocusChange: source.handler(['onFocusChange'], (VoidCallback trigger) => (bool focus) => trigger()),
+      autofocus: source.v<bool>(['autofocus']) ?? false,
+      hoverDuration:
+          ArgumentDecoders.duration(source, ['hoverDuration'], context),
+      child: source.child(['child']),
```
[the remark this answers is not in the document]
Comment (peixinli) [reply packages/rfw/lib/src/flutter/material_widgets.dart]: Fixed.

File: packages/rfw/lib/src/flutter/material_widgets.dart [hunk 5]
```diff
@@ -337,6 +340,42 @@ Map<String, LocalWidgetBuilder> get _materialWidgetsDefinitions => <String, Loca
...
+    );
+  },
+
   'InkWell': (BuildContext context, DataSource source) {
     // not implemented: onHighlightChanged, onHover; mouseCursor; focusColor, hoverColor, highlightColor, overlayColor, splashColor; splashFactory; focusNode, onFocusChange
```
[the remark this answers is not in the document]
Comment (peixinli) [reply packages/rfw/lib/src/flutter/material_widgets.dart]: Done!

File: packages/rfw/lib/src/flutter/material_widgets.dart [hunk 6]
```diff
@@ -337,6 +340,42 @@ Map<String, LocalWidgetBuilder> get _materialWidgetsDefinitions => <String, Loca
     );
   },
 
+  'InkResponse': (BuildContext context, DataSource source) {
+    // not implemented: mouseCursor, overlayColor, splashFactory, focusNode.
+    return InkResponse(
+      onTap: source.voidHandler(['onTap']),
+      onTapDown: source.handler(['onTapDown'], (VoidCallback trigger) => (TapDownDetails details) => trigger()),
```
[the remark this answers is not in the document]
Comment (peixinli) [reply packages/rfw/lib/src/flutter/material_widgets.dart:346]: Filed the issue #144175. We should expose the details for all the tap events.

**Pull request merged**